alica_engine/include/engine/BasicPlan.h

```diff
@@ -24,9 +25,11 @@ class BasicPlan
     void setEngine(AlicaEngine* engine) { _ae = engine; }
     void setRunningPlan(RunningPlan* rp) { _context = rp; }
     void setConfiguration(const Configuration* conf) { _configuration = conf; }
+    void setName(const std::string& name) { _name = name; }
 
     AlicaTime getInterval() { return _msInterval; }
     void setInterval(int32_t msInterval) { _msInterval = AlicaTime::milliseconds(msInterval); }
+    IAlicaTrace& getTrace() { return *_trace; };
```

Comment:
Why is this public? If the plan is to use this for adding child dependencies + You can return a simple std::string context  like in TraceContext.msg. rather than returning  reference to trace.

Reply:
this is for the application to set custom logs & tags inside the behaviour/plan's implementation, so we need to return the full trace object.

Reply:
but this will add one more nesting in the trace tree which could make trace look more visually complicated

Reply:
ah I understand, just make this protected.